### PR TITLE
Add codex session syncing

### PR DESCRIPTION
**Pull request by jqdsouza**
> [!NOTE]
> **Medium Risk**
> New local file ingestion and runtime log writes could mis-map or duplicate telemetry if Codex formats change, but collection is poll-only and does not intercept tool execution.
> 
> **Overview**
> Adds **`beacon endpoint codex`** with **`sync`** and **`status`** so Beacon can backfill endpoint telemetry from Codex CLI’s committed JSONL under `~/.codex/sessions`, complementing live OTLP capture when sessions ran without Beacon or are historical.
> 
> The new **`codexsession`** package discovers rollout files, decodes JSONL (including partial tails and oversized lines), maps transcripts to endpoint events (`prompt.submitted`, `tool.invoked`, `command.executed`, `token.usage`, etc.) with **`harness.collection_method=poll`**, deterministic event IDs, and incremental **cursor state** (`codex.json`) so sweeps only append new lines. **`sync`** supports dry-run **`--print`**, optional **`--watch`**, and writes via the existing runtime log writer; **`status`** reports per-session collection progress vs. the cursor.
> 
> Token usage mapping prefers **`token_usage_record`** when present and deduplicates noisy **`token_count`** snapshots (deltas, session-scoped totals, one event per model completion). Tests cover listing, mapping, cursor behavior, and token edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 41868b590e8d3e150b1051086559aaf38648bfa9. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: cli/beacon/internal/codexsession/mapper.go [hunk 1]
```diff
@@ -0,0 +1,662 @@
...
+	if turnID != "" && m.emittedTokenCountTurns[turnID] {
+		return
+	}
+	usage := usageFromCodex(firstUsage(msg.Info.LastTokenUsage, msg.Info.TotalTokenUsage))
+	if usage == nil {
+		return
+	}
+	if turnID != "" {
+		m.emittedTokenCountTurns[turnID] = true
+	}
```
[the remark this answers is not in the document]
Comment (jqdsouza) [reply cli/beacon/internal/codexsession/mapper.go]: Fixed in 80eefb9 — and it also replaces the last-wins attempt in bb335af, which had the mirror-image bug (a turn of 20 + 30 output tokens reported 30).

`consumeTokenCount` no longer keeps one snapshot per turn. It deduplicates on the cumulative `total_token_usage` instead: an unchanged total means Codex re-emitted without a new model completion and is skipped, a grown total is a new completion and is emitted from its `last_token_usage`. The usage reported for a turn now adds up to that turn's cumulative total.

Also in the same commit:
- `usageFromCodex` returns nil for an all-zero snapshot, so an empty early `token_count` neither emits an event nor hides later real usage (this covers the `token_usage_record` path too).
- A snapshot with no `last_token_usage` reports the growth of the cumulative total rather than the running total, so the fallback can't inflate.
- Totals are tracked for records below the cursor as well, so a resumed sweep deduplicates against what the previous sweep already wrote.

`token_usage_record` still wins for turns that have one. Pinned by `TestTokenCountCoversEveryCompletionInAMultiCallTurn`, `TestTokenCountEmitsEveryModelCompletionOnce`, `TestTokenCountWithoutLastUsageReportsDelta`, and `TestTokenCountYieldsToTokenUsageRecord`.

---
_Generated by [Claude Code](https://claude.ai/code)_
